graph/pull_v1.go

```diff
@@ -138,16 +138,20 @@ func (p *v1Puller) pullRepository(askedTag string) error {
 			}
 
 			// ensure no two downloads of the same image happen at the same time
-			broadcaster, found := p.poolAdd("pull", "img:"+img.ID)
+			poolKey := "img:" + img.ID
+			broadcaster, found := p.poolAdd("pull", poolKey)
+			broadcaster.Add(out)
 			if found {
-				broadcaster.Add(out)
-				broadcaster.Wait()
+				err = broadcaster.Wait()
```

Comment:
Yes, I'll make this change. It was originally like this so that the `err == nil` case would display a "Download complete" message, but this message is redundant with the one that gets produced by the actual download.